sherpa/astro/tests/test_astro_data2.py

```diff
@@ -816,3 +816,46 @@ def test_pha_invalid_units(invalid, make_test_pha):
         pha.units = invalid
 
     assert str(de.value) == f"unknown quantity: '{invalid}'"
+
+
+def test_pha_reverse_sorted():
+    '''This case for a PHA set with reverse sorted channels is a little
+    non-sensical, but it is a very easy way to hit the problem in #1163.
+
+    Really, we want to test that with a larger test that reads real XMM
+    data, which uses reverse energy ordering (see next test).
+
+    So, if thise tests starts to fail, but the next one passes, this one can
+    probably be removed. It's here now because it's fast, easy, and does not
+    require the large test data.
+    '''
+    d = DataPHA('x', np.array([4, 3, 2, 1, 0]), np.array([5, 4, 3, 2, 1]))
+    d.notice(2, 4)
+    assert len(d.get_dep(filter=True)) == 3
+
+
+@pytest.fixture
+def read_test_image(make_data_path):
+    from sherpa.astro.io import read_image
+    filename = 'acisf07999_000N001_r0035_regevt3_srcimg.fits'
+    d = read_image(make_data_path(filename))
+    d.name = 'test.img'
+    return d
+
+
+@requires_fits
+@requires_data
+def test_xmmrgs_notice(make_data_path):
+    '''Test that notice and ignore works on XMMRGS dataset, which is
+    ordered in increasing wavelength, not energy'''
+    from sherpa.astro.ui.utils import Session
+    session = Session()
+    session.load_data(make_data_path('xmmrgs/P0112880201R1S004SRSPEC1003.FTZ'))
+    session.load_rmf(make_data_path('xmmrgs/P0112880201R1S004RSPMAT1003.FTZ'))
```

Comment:
I note you add four files to `sherpatest` but only seem to use 2 here. Are the header keywords set up so that the other files automatically loaded here? Or is the plan to use these files with other tests yet-to-be-written?

Reply:
The other two are a background spectra. They are both relatively small files. I don't have any other tests ready to go on my disk, but in the interest of making the data reusable for other tests, I wanted to have the minimum viable number of data files so that we *could* be used for other tests, e.g. background subtraction, plotting with background spectra etc. It's going to be a pain to find the exact matching versions in the archive again (because XMM also goes through repro cycles, which may add keywords etc.), so I think we should invest the extra 39 kB to hae then when we need them.